netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -1944,20 +2023,18 @@ net_ebpf_extension_sock_addr_redirect_connection_classify(
             incoming_metadata_values->transportEndpointHandle, sock_addr_ctx);
     }
 
-    // Callout at CONNECT_REDIRECT layer always returns WFP action PERMIT.
-    // If the eBPF program was invoked and it returned a REJECT verdict, it would be enforced by the callout at
-    // AUTH_CONNECT layer further downstream.
-
-    classify_output->actionType = FWP_ACTION_PERMIT;
+    // Callout at CONNECT_REDIRECT layer always returns WFP action PERMIT / CONTINUE.
+    // If the connection was redirected or blocked, make the action TERMINATING.
+    if (reauthorization || redirected || verdict == BPF_SOCK_ADDR_VERDICT_REJECT) {
```

Comment:
I am not sure I follow this part - if verdict == VERDICT_REJECT, then we set the action to PERMIT?

Reply:
Oh nevermind, I think I follow - you cannot block at connect redirect layer.